recipes-kernel/linux/files/0001-Allowing-sdhci_msm-to-compile-as-a-loadable-module.patch

```diff
@@ -0,0 +1,82 @@
+From 76b69fe765c2c7f2c8f1ebeac4d7290fb6d08db4 Mon Sep 17 00:00:00 2001
+From: Zahid Chowdhury <zchowdhury@sierrawireless.com>
+Date: Wed, 28 Aug 2019 09:47:54 -0700
+Subject: [PATCH] Allowing sdhci_msm to compile as a loadable module
+
+Change-Id: Ia9878973e0de413f44c008924c5b9b696c39a2fc
+---
+ drivers/mmc/core/host.c      |  3 +++
+ drivers/mmc/host/sdhci-msm.c | 17 +++++++++++++++++
+ 2 files changed, 20 insertions(+)
+
+diff --git a/drivers/mmc/core/host.c b/drivers/mmc/core/host.c
+index ccaffd2b7ebd..9c0b4511cf25 100644
+--- a/drivers/mmc/core/host.c
++++ b/drivers/mmc/core/host.c
+@@ -206,6 +206,9 @@ bool mmc_host_may_gate_card(struct mmc_card *card)
+ 	return !(card->quirks & MMC_QUIRK_BROKEN_CLK_GATING);
+ }
+ 
++/* SWI - added to allow sdhci-msm to become a loadable module */
++EXPORT_SYMBOL(mmc_host_may_gate_card);
++
+ /**
+  *	mmc_host_clk_release - gate off hardware MCI clocks
+  *	@host: host to gate.
+diff --git a/drivers/mmc/host/sdhci-msm.c b/drivers/mmc/host/sdhci-msm.c
+index 8a2e27870d8d..8a3f6d91f699 100644
+--- a/drivers/mmc/host/sdhci-msm.c
++++ b/drivers/mmc/host/sdhci-msm.c
+@@ -3908,6 +3908,11 @@ static void sdhci_msm_cmdq_init(struct sdhci_host *host,
+ }
+ #endif
+ 
++/* SWI - making sdhci_msm into a a loadable module requires probe eMMC as a boot
++ * device to be commented out and then this routine becomes unused and gcc
++ * complains.
++ */
++#if 0
```

Comment:
Shouldn't this use a KCONFIG instead of `0`?

Reply:
No existing Kconfig existed for eMMC as a boot device in this old kernel. Definitely would be the way to go if it existed. Don't want to create a new Kconfig for just current dead code.